test/e2e/daemon_set.go

```diff
@@ -427,12 +427,27 @@ func checkRunningOnAllNodes(f *framework.Framework, selector map[string]string)
 		framework.ExpectNoError(err)
 		nodeNames := make([]string, 0)
 		for _, node := range nodeList.Items {
+			if taintedNoSchedule(node) {
```

Comment:
I didn't realize `isNodeTainted` existed.

DaemonSet can and should launch pods on nodes that are NotReady or Unschedulable.